airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlDebeziumConverter.java

```diff
@@ -129,9 +125,21 @@ private void registerDatetime(final RelationalColumn field,
           if (Objects.isNull(input)) {
             return DebeziumConverterUtils.convertDefaultValue(field);
           }
+          if (input instanceof final Timestamp d) {
+            final LocalDateTime localDateTime = d.toLocalDateTime();
+            return localDateTime.format(DateTimeFormatter.ofPattern(DATETIME_FORMAT_MICROSECONDS));
+          }
+
+          if (input instanceof final Long d) {
```

Comment:
Is it the case that for Mssql that all timestamp 'numbers' are _always_ with nanosecond precision?

Reply:
We're only seeing this Long input when values are converted as part of acquiring schema history.
Dates on actual row records are still seen as incoming `TimeStamp`. None of this is documented anywhere as far as I can tell.

We need to take a close look at some point as this area is uneven in terms of implementation.
And [some of the code](https://github.com/airbytehq/airbyte/commit/56db8065e9fbfbf054303c9d1a8c9694547dd02a) here is almost three years old.